metis-dereference/metis-dereference-service/src/main/java/eu/europeana/metis/dereference/service/MongoDereferenceService.java

```diff
@@ -129,16 +134,43 @@ public EnrichmentResultList dereference(String resourceId)
 
     // Perform the breadth-first search to search for broader terms (if needed).
     final int iterations = resource.getRight().getIterations();
-    final Collection<EnrichmentBase> result;
+    final Map<String, EnrichmentBase> result;
     if (iterations > 0) {
       result = GraphUtils.breadthFirstSearch(resourceId, resource.getLeft(),
           resource.getRight().getIterations(), valueResolver, this::extractBroaderResources);
     } else {
-      result = Collections.singleton(resource.getLeft());
+      result = Collections.singletonMap(resourceId, resource.getLeft());
     }
 
+    // Get links from dereferenced objects and try to enrich them. Get: 1) entities that have any of
```

Comment:
Does this comment block need update? I read it a bit differently on the ticket and in code below seems much simpler..?

Reply:
Well, let's say that I added it like this precisely because the ticket is not 100% clear. I'm going by the latest comment of Valentine in the ticket, which seems the most clear to me. And it is this that should be implemented by the code below the comment. I would suggest leaving the wordy comment in place, just so that we know what the deceptively simple-looking code is actually doing behind the scenes.